site-src/geps/gep-1867.md

```diff
@@ -0,0 +1,137 @@
+to a GatewayClass.
+
+This GEP will explain why that is not sufficient to meet common use cases, and introduce a new field - `infrastructure` - to address these cases.
+
+Related discussions:
+* [Support cluster-local Gateways](https://github.com/kubernetes-sigs/gateway-api/discussions/1247)
+* [Scaling Gateway Resources](https://github.com/kubernetes-sigs/gateway-api/discussions/1355)
+* [Manual deployments](https://github.com/kubernetes-sigs/gateway-api/issues/1687)
+* [Merging Gateways](https://github.com/kubernetes-sigs/gateway-api/pull/1863)
+* [In Cluster Gateway Deployments])(https://github.com/kubernetes-sigs/gateway-api/pull/1757)
+
+## Goals
+
+* Provide the ability to configure arbitrary (implementation specific) attributes about a specific Gateway.
+* Provide the ability to configure a standardized set attributes about a specific Gateway.
+
+## Challenges with GatewayClass
+
+`GatewayClass.spec.parametersRef` is the existing mechanism to configure arbitrary fields on a Gateway.
+However, this introduces operational challenges when configuring Gateways.
+
+### Scope
+
+GatewayClass is a cluster scoped resource. This effectively limits customization to either:
+1. Cluster admin pre-provisions a few classes, lower-privilege Gateway admins select them.
+2. Gateway admin and Cluster admin are the same.
+
+Case 1 works great when there are only a handful of customizations needed.
+For example, you could have a "public" vs "private" class, or a "big" and "small" class.
+This doesn't scale beyond simple cases, however; even the example above requires 4 classes and exponentially grows as more use cases are needed.
+
+Case 2 works fine, but greatly limits use cases, such as [Support namespace scoped implementations](https://github.com/kubernetes-sigs/gateway-api/issues/567).
+Rather than having 1 big ingress for the entire cluster, it is common for clusters to have multiple smaller gateways ([example](https://istio.io/latest/docs/setup/additional-setup/gateway/#gateway-deployment-topologies).
+This can be A gateway per app, per namespace,  per "tenant", or any mix of these.
+
+### Custom Resource
+
+`parametersRef` is entirely a generic implementation-specific meaning.
+This means users will either need a customer CRD or use untyped resources like ConfigMap.
+Neither of these have any consistency between implementations.
+While there will always be some vendor-specific requirements, there are also a number of configuration aspects of a Gateway that are common between implementations.
+However, these cannot currently be expressed in a vendor-neutral way.
+
+Additionally, there is hesitancy to use a CRD (which leads to CRD proliferation), which pushes users towards untyped ConfigMaps which are not much better than annotations. The scoping, as mentioned above, is also a bit awkward of a cluster scoped resource pointing to a namespaced object.
+
```

Comment:
While I'm not saying this issue is without merit, I think it's the least compelling of the 4 issues brought up _to me_. I personally would just remove it, as I think the rest stands fine without it and this one is slightly more subjective:

```suggestion
```

However I don't consider this comment blocking if you feel really strongly it's an important part of the decision making.

Reply:
I also do think its the least meaningful but it is important IMO. Most of gateway-api is about balancing extension points vs 1p fields, so this addresses that aspect